### ISSUE TITLE
ToDo: experiment with using CFM's as modules

### ISSUE THREAD

**Issue by jfrux** [foundrycf/foundry#20]
We all know that cfinclude can create a "mixin" style environment for any component.
But cfinclude is not very elegant and doesn't lend itself well to modular development.

So my theory is, if we standardize cfm-based foundry modules, or possibly even not standardize it but just allow it, we can use require("module-id") and it would grab a cfm file and include it into a component or a page without the need of a file path or even care about that kind of thing.

cfinclude would always be more efficient when including your own files but require() just gives you one more option for modular development.

Components somewhat lock you down to a particular type of object or structure, with cfm-based modules, you could do different things.

**Comment by russplaysguitar** [comment]
Don't forget about cfmodule! Not sure where exactly that would fit in to the idea...

**Comment by jfrux** [comment]
Fair enough, actually, that's perfect... we would just execute it as a customtag (probably) or use that as the syntax of sorts.

I'll look into it more with that in mind.

Thanks!

**Comment by russplaysguitar** [comment]
Not sure how familiar you are with cfmodule, but this might be helpful info: 
- cfmodule can be used to include arbitrary cfm files, not just just custom tags
- cfmodule templates are executed within their own scope (as opposed to include, which executes within the same scope as the calling page)
- cfmodule will call the template twice if you include the self-closing xml syntax `<cfmodule ... />` or write the open/close tags `<cfmodule ...> ... </cfmodule>`, but only once if you don't self-close a single tag (`<cfmodule ...>`)
